src-java/server42/server42-stats/src/main/java/org/openkilda/server42/stats/zeromq/StatsCollector.java

```diff
@@ -60,25 +73,33 @@ public void run() {
         sessionId = RandomStringUtils.randomAlphanumeric(8);
         log.info("started with session id {}", sessionId);
         while (!isInterrupted()) {
-            try (ZContext context = new ZContext()) {
-                Socket server = context.createSocket(ZMQ.PULL);
-                try {
-                    server.connect(connectEndpoint);
-                    while (!isInterrupted()) {
-                        byte[] recv = server.recv();
-                        log.debug("stats recived");
-                        handleInput(recv);
+            try (Socket server = context.createSocket(ZMQ.PULL)) {
+                server.setReceiveTimeOut(1000);
+                server.connect(connectEndpoint);
+                log.info("connect to {}", connectEndpoint);
+                while (!isInterrupted()) {
+                    byte[] recv = server.recv();
+                    log.debug("recv {}", recv);
+                    if (recv == null && server.errno() == ZError.EAGAIN) {
```

Comment:
why is it break for eagain

Reply:
That is timeout. I didn't find documentation but I found src https://github.com/zeromq/jeromq/blob/master/src/main/java/zmq/SocketBase.java#L850